### PR TITLE
Adding Dynamic Names

### PR DESCRIPTION
**Pull request by ghost**
I need a review, let me know if this is okay and if I need to build this with Ronn.  
I will be pretty busy these next few days until September so feel free to commit changes in my fork by yourself if you feel to. I will try my best to be here. :)

### CODE REVIEW

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: The manpage, at least in its current form, is tag-oriented. Dotted names and lambdas are discussed as sub-features of variable tags and block tags. For consistency, I would suggest similarly treating dynamic names as a subsection of partial tags. That means formatting the title in bold, instead of as a level 3 heading:

```suggestion
**Dynamic Names**
```

Conveniently, you already placed this section exactly where it needs to be in order to discuss it as a subfeature of partial tags.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: This is very technical and formal compared to the rest of the manpage. Please try reformulating it more loosely in terms of what it's good for and how you can use it.

It would also be good to mention that dynamic names are an optional part of the specification. Perhaps see the lambdas subsection of the Block tags section for some inspiration.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron:441]: I think these examples are fine the way they are.

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
Thanks a lot! Sorry for the delayed review, for some reason I was not watching all activity on my own fork yet. That's fixed now.

Besides my comments below, I have one request:

- There is a paragraph near the top of the manpage where it mentions the spec. That should be updated to say that it reflects version 1.3.0 of the spec.

and one suggestion:

- Maybe briefly mention dynamic names again in the section on Parent tags. Compare to how I document dotted names in the Variables section, and then briefly mention them again in the Blocks section.

> I need a review, let me know if this is okay and if I need to build this with Ronn.

It's convenient for me if you build it, but I know this can be challenging. If it proves too challenging, I don't mind doing it myself.

> I will be pretty busy these next few days until September so feel free to commit changes in my fork by yourself if you feel to. I will try my best to be here. :)

I might do that at some point, if I think I can make the changes much sooner than you (which is not yet the case), but for now, I'll just leave it at documenting my thoughts. :-)

**Comment by ghost** [comment]
> Thanks a lot! Sorry for the delayed review, for some reason I was not watching all activity on my own fork yet. That's fixed now.

No worries. I have been kind of busy lately but now I'm back at it again!

> Besides my comments below, I have one request:
> 
>     * There is a paragraph near the top of the manpage where it mentions the spec. That should be updated to say that it reflects version 1.3.0 of the spec.
> and one suggestion:
> 
>     * Maybe briefly mention dynamic names again in the section on Parent tags. Compare to how I document dotted names in the Variables section, and then briefly mention them again in the Blocks section.

I got it. I am working on these right now.

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: Right!

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: Okay I got it!

**Comment by ghost** [comment]
Okay I tried to make the description less technical and verbose.  
I am not sure whether the way I mentioned Dynamic Names in the Blocks and Parents paragraphs is okay, what was your idea on that?

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: "optional" is a bit ambiguous here. It can be interpreted as "optional part of the spec" but also as "optional to use". I suggest adding some words to clarify that not every Mustache engine might implement this notation.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: Perhaps compare this to the hypothetical `{{>{{dynamic}}}}` to further clarify what it does.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: I think this sentence can be left out. The manpage doesn't spend much attention to this corner case for most of the other tags, either.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: In the Blocks section, this remark is probably just confusing.

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
> I am not sure whether the way I mentioned Dynamic Names in the Blocks and Parents paragraphs is okay, what was your idea on that?

Your current text seems to be based on the observation that inheritance and dynamic partials are to some extent alternatives. Since most manpage readers will be unaware of that, I agree it's probably not very helpful.

I meant that some implementations (at least mine, possibly also the ones by @gasche or @bobthecow) allow using dynamic names inside parent tags:

```
{{<*path.to.parentName}}{{/*path.to.parentName}}
```

So it may be useful to mention that this is an option in some implementations, similar to how the Section section mentions that dotted names are an option (after the Variables section has already explained at length what dotted names do).

**Comment by ghost** [comment]
Okay I'm back at it again!  
Sorry for the delay I was emotionally struggling lol  
Now it's time to work 💪  
Pushing some changes soon

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: got it!

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: I think that including wrong examples may be confusing

**Comment by jgonggrijp** [comment]
> Sorry for the delay I was emotionally struggling lol

Oh, no need to apologize! I hope the struggle was not because of this pull request? In any case, best wishes!

(I will review next.)

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: I'm used to calling it "interpolation tag", too, but earlier on in the manpage it's called "variable", so for clarity best to go with that here.

```suggestion
the same notation and the same resolution as in an variable tag. That is
```

File: man/mustache.5.ron
Comment (jgonggrijp) [reply man/mustache.5.ron]: You are quite right that it could be confusing. Very friendly of you to go with it anyway! I still think the hypothetical nested tag can also serve to clarify the concept, but please feel absolutely welcome to add an additional remark, something like `(which is not allowed!)` just before or after the forbidden example.

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]: You can be more concise here, since the optionality of the feature was already mentioned in the previous section.

```suggestion
Some mustache implementations may allow the use of Dynamic Names in
parent tags, similar to dynamic names in partials. Here's an example of
how Dynamic Names in parent tags work.
```

File: man/mustache.5.ron
Comment (jgonggrijp) [inline man/mustache.5.ron]:
```suggestion
Templates:
```

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
I'm quite happy with this! At this point, I feel I'm just nitpicking on details.

File: man/mustache.5.ron
Comment (jgonggrijp) [reply man/mustache.5.ron]: (I only added the final 's' to make it plural. No idea why GitHub chooses to highlight everything after the capital 'T' here...)

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: Right! That looks nice.

File: man/mustache.5.ron
Comment (ghost) [reply man/mustache.5.ron]: > (I only added the final 's' to make it plural. No idea why GitHub chooses to highlight everything after the capital 'T' here...)

pretty weird yeah.  
I'm committing this suggestion now by the way!

**Comment by ghost** [comment]
> Oh, no need to apologize! I hope the struggle was not because of this pull request? In any case, best wishes!

Noooo noo absolutely not! I like working with you!  
It was due a girl.  
Ehh.. such is love xD
But I feel better now, thank you


Okay so now I have committed all the changes you suggested and the only thing I have left is to add an additional remark to that forbidden example we were discussing earlier.  
I'm committing in a few minutes and then I think I'm done!

**Review by jgonggrijp — APPROVED** [review]
Great job, thanks! Feel free to compile the thing, or not. Otherwise I'll do it just before merging (that will be tomorrow at the earliest, I'm going to bed now).

**Comment by ghost** [comment]
> Great job, thanks! Feel free to compile the thing, or not. Otherwise I'll do it just before merging (that will be tomorrow at the earliest, I'm going to bed now).

Awesome!  
Okay I managed to compile the thing, you should take a look at my last commit, if something went wrong I can revert it.  
PS. I had to edit the Rakefile too since it was looking for `*.ronn` files meanwhile we have `*.ron`.  
I'm also going to bed now, goodnight! 😴

**Comment by bobthecow** [comment]
> I meant that some implementations (at least mine, possibly also the ones by gasche or bobthecow) allow using dynamic names inside parent tags

Yup, my implementation does as well.

**Comment by jgonggrijp** [comment]
@anomal00us could you change the last commit so it omits the mustache.1 files? The annoying thing about the rebuild is that it makes those files seem as if they were updated on the date of the latest build, even though the content is exactly the same as... *many* years ago.

**Comment by ghost** [comment]
> @anomal00us could you change the last commit so it omits the mustache.1 files? The annoying thing about the rebuild is that it makes those files seem as if they were updated on the date of the latest build, even though the content is exactly the same as... _many_ years ago.

Sure!

**Comment by jgonggrijp** [comment]
Thanks again @anomal00us !

**Comment by ghost** [comment]
You're welcome, it has been nice to work with you :)

**Pull request merged**
